src/elements/global-header/global-header.js

```diff
@@ -22,18 +17,22 @@ export const GlobalHeader = connect({
           <li><Link name='users' activeClass='active-link'>Dashboard App</Link></li>
           <li><button onClick={increment}>+</button></li>
         </ul>
-
       </div>
 
       <div className='user-actions'>
-
         <Avatar
           src='/images/_temp/avatar.png'
           fullName='John Smith'
-
         />
-
       </div>
+
     </div>
   </header>
-))
+
+export const GlobalHeader = () => {
+  const clicks = useMappedState(store, ({ clicks }) => clicks)
+
+  const increment = () => store.setState({ clicks: clicks + 1 })
+
+  return Base({ clicks, increment })
```

Comment:
Do you find a benefit to breaking out a `Base` component in this instance?

Personally, I prefer when the source of the data is closer to the rendering of the data. And, with the advent of hooks, it's easy to abstract away logic when a component definition gets too busy.

Reply:
I dislike Base, was only using it for the sake of consistency. Happy to remove and cease using.

Reply:
In this case it also made for a smaller and more legible diff (no change to indentation, only relevant changes highlighted)